dataland-frontend/src/components/pages/SearchCompaniesForFrameworkData.vue

```diff
@@ -134,38 +206,108 @@ export default defineComponent({
         }
       }
     },
-    scanQueryParams() {
-      let queryFrameworks = this.route.query.frameworks;
-      if (queryFrameworks) {
-        if (typeof queryFrameworks === "string" && queryFrameworks !== "") {
-          this.currentFilteredFrameworks = [queryFrameworks as DataTypeEnum];
-        } else if (Array.isArray(queryFrameworks)) {
-          this.currentFilteredFrameworks = queryFrameworks as Array<DataTypeEnum>;
-        }
+    getQueryFrameworks(route: RouteLocationNormalizedLoaded): Array<DataTypeEnum> {
+      let queryFrameworks = route.query.framework;
+      if (queryFrameworks !== undefined) {
+        const allowedDataTypeEnumValues = Object.values(DataTypeEnum) as Array<string>;
+        return parseQueryParamArray(queryFrameworks).filter((it) =>
+          allowedDataTypeEnumValues.includes(it)
+        ) as Array<DataTypeEnum>;
       } else {
-        this.currentFilteredFrameworks = Object.values(DataTypeEnum);
+        return Object.values(DataTypeEnum);
       }
```

Comment:
why not declaring the allowedDataTypeEnumValues once, and then return it after the else?